app/src/main/java/lang_import/org/app/ArticleActivity.kt

```diff
@@ -96,40 +126,93 @@ class ArticleActivity : AppCompatActivity() {
         return fixImg(txt)
     }
 
+    var status: CharSequence = ""
+        set(d) {
+            val view = findViewById<TextView>(R.id.article_progress_status)
+            runOnUiThread {
+                view.text = d
+                findViewById<ViewGroup>(R.id.article_progress).visibility = View.VISIBLE
+            }
+
+            field = d
+        }
 
-    suspend fun importLang(txt: String): String {
-        var rep = txt
+    fun done() {
+        runOnUiThread {
+            findViewById<ViewGroup>(R.id.article_progress).visibility = View.GONE
+        }
+    }
+
+    suspend fun importLang(rawText: String): String {
+        // Get only text-lines from strict list of tags (skip hrefs and tag attrs)
+        var goodLines = mutableListOf<AcrticlePart>()
+        val nodesLst = NodeIter(Jsoup.parse(rawText).body(), mutableListOf())
+
+        // Collect Nodes from informers
+        for (node in nodesLst) {
+            val articleObj = AcrticlePart()
```

Comment:
`AcrticlePart` - typo mistake?
We can init ArticlePart as immutable object here